File: locust_benchmark/benchmark.py
import time
import warnings
from locust import HttpUser, task, between

warnings.filterwarnings("ignore")

class QuickstartUser(HttpUser):
    wait_time = between(1, 2.5)

    @task
    def status(self):
        self.client.get("/api/status", verify=False)

    @task
    def healthcheck(self):
        self.client.get("/api/healthcheck", verify=False)

    @task
    def updater(self):
        self.client.get("/api/updater", verify=False)

    @task
    def updater_wrong_data(self):
        self.client.post("/api/updater", verify=False, json={})

    @task
    def updater_correct_data(self):
        self.client.post("/api/updater", verify=False, json={
            "database": "locust",
            "schema": {
                "my": "schema"
            },
            "interval": 60,
            "source": "https://ipinfo.io/json",
            "method": "GET",
            "timeout": 30
        })

    @task
    def data(self):
        self.client.post("/api/data", verify=False, json={
            "database": "locust"
        })

    @task
    def data_with_data(self):
        self.client.post("/api/data", verify=False, json={
            "database": "locust",
            "quantity": 50
        })

File: k6/benchmark.js
import http from 'k6/http';
import { sleep } from 'k6';

export const options = {
    duration: '1m',
    vus: 150,
    insecureSkipTLSVerify: false,
};

export default function () {
    // Status and heartbeat
    http.get('https://localhost/api/status');
    http.get('https://localhost/api/healthcheck');

    // Updaters
    http.get('https://localhost/api/updater');
    http.post('https://localhost/api/updater', {});
    http.post('https://localhost/api/updater', {
        database: "k6",
        schema: {
            my: "schema"
        },
        interval: 60,
        source: "https://ipinfo.io/json",
        method: "GET",
        timeout: 30
    });

    // Data
    http.post('https://localhost/api/data', {database: "k6"})
    http.post('https://localhost/api/data', {database: "k6", quantity: 50})

    // 404
    http.get('https://localhost/api/yikes');

    sleep(1);
}